lib/model/folder.go

```diff
@@ -921,6 +935,40 @@ func (f *folder) emitDiskChangeEvents(fs []protocol.FileInfo, typeOfEvent events
 	}
 }
 
+func (f *folder) forceRescan() {
+	paths := make([]string, 0, len(f.forcedRescanPaths))
+
+	f.forcedRescanPathsMut.Lock()
+	for path := range f.forcedRescanPaths {
+		paths = append(paths, path)
+		delete(f.forcedRescanPaths, path)
```

Comment:
I suggest just throwing away the map and creating a new one after the loop, as I don't think map structures shrink once they've grown.